Source/Scene/BatchTable.js

```diff
@@ -349,13 +504,34 @@ define([
                '    float centerX = batchTextureStep.y; \n' +
                '    float stepY = batchTextureStep.z; \n' +
                '    float centerY = batchTextureStep.w; \n' +
-               '    float numberOfAttributes = float('+ numberOfAttributes + '); \n' +
+               '    float numberOfAttributes = float('+ stride + '); \n' +
                '    float xId = mod(batchId * numberOfAttributes, batchTextureDimensions.x); \n' +
                '    float yId = floor(batchId * numberOfAttributes / batchTextureDimensions.x); \n' +
                '    return vec2(centerX + (xId * stepX), 1.0 - (centerY + (yId * stepY))); \n' +
                '} \n';
     }
 
+    function getGlslUnpackFloat(batchTable) {
+        if (!batchTable._packFloats) {
+            return '';
+        }
+
+        return 'float unpackFloat(vec4 value) \n' +
```

Comment:
I wouldn't move this because it mirrors `unpackFloat`. If we could use the packing/unpacking code elsewhere, I'd be OK with it. Can you think of another case where we would do something like this?